### DISCUSSION TITLE
Help installing on lenovo c330

### DISCUSSION THREAD

**Discussion by cdd100** [Maccraft123/Cadmium#143] [General]
Stupid question, but where is the download for lenovo c330.

Thank you.

**Comment by deiol** [comment]
The releases are no longer device-specific, so you'll probably want to try out one of the v0.4.0-pre2 images

**Reply by cdd100** [reply]
Thank you, I will update you if it works.

**Comment by cdd100** [comment]
I have tried both the arm64 and arm32 versions, neither of them have worked but my Chromebook is listed in the supported hardware list. Please help! The farthest I have gotten is a black screen when I press ctrl+u. It doesn't change. The imager I am using is the raspberry pi imager. I am new to getting Linux on arm Chromebooks.

**Reply by ddcampayo** [reply]
Hi, I have tried the arm64 debian version on a c330, but it's not working. I sign as guest, open a terminal with ctrl+alt+forward, go root typing "root", then "enable_dev_usb_root", then "reboot". In the screen that results, with an exclamation mark, and "OS verification is disabled", all I get is two ugly beeps, after a while, and another one if I press ctrl+U.

Trying arm32, but I think the system is 32. Could it be another version? void??

**Reply by chinh4** [reply]
the support hasn't been implemented when the images were built, so you're gonna have to build it yourself, also it's arm64

**Reply by ddcampayo** [reply]
OK, I'll try that. It's not clear to me which option is correct for the C330:

# One of: bob, duet, kevin, minnie, speedy, lazor

duet ?? (that one's a Lenovo, at least)

**Reply by chinh4** [reply]
why are you using a old config script?

**Reply by ddcampayo** [reply]
Ah, you are right. Trying the prerelease now ...

**Reply by ddcampayo** [reply]
Hm. The standard config leads to errors regarding "/drivers/staging/android". That directory does not exist, the closest is "tmp/linux-arm64/drivers/staging", but that one has no "android" subdir.

Error:

`Applying /home/daniel/Downloads/Cadmium-0.4.0-pre2/kernel/patches/common.android-enable-building-ashmem-and-binder-as-modules.patch
patching file drivers/android/Kconfig
Hunk #1 succeeded at 2 with fuzz 1 (offset -7 lines).
patching file drivers/android/Makefile
patching file drivers/android/binder_alloc.c
can't find file to patch at input line 69
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|Index: b/drivers/staging/android/Kconfig
|===================================================================
|--- a/drivers/staging/android/Kconfig
|+++ b/drivers/staging/android/Kconfig
--------------------------
`

**Reply by ddcampayo** [reply]
I see this issue has arisen earlier https://devscope.io/code/Maccraft123/Cadmium/issues/124

**Reply by chinh4** [reply]
> I see this issue has arisen earlier https://devscope.io/code/Maccraft123/Cadmium/issues/124
> 

https://github.com/chinh4thepro/Cadmium-CI-CD/actions/runs/3140314281

you can use my build

**Reply by ddcampayo** [reply]
I tried to follow your steps, and I definitely reached further. But building stops at this stage:

```
I: Extracting libpam-modules-bin...
E: Tried to extract package, but file already exists. Exit...

```
Trying debian stable now, see if that works.

Best,

Daniel

**Reply by chinh4** [reply]
> I tried to follow your steps, and I definitely reached further. But building stops at this stage:
> 
> ```
> I: Extracting libpam-modules-bin...
> E: Tried to extract package, but file already exists. Exit...
> 
> ```
> Trying debian stable now, see if that works.
> 
> Best,
> 
> Daniel
> 
> 

Same thing happens to me, probably an issue with debootstrap

**Reply by cdd100** [reply]
I understand this is off-topic, but Arch Linux arm wont load. I get to the bootscreen but it repeats the same commands. Is there any way to fix this?

**Reply by chinh4** [reply]
> I understand this is off-topic, but Arch Linux arm wont load. I get to the bootscreen but it repeats the same commands. Is there any way to fix this?

I have recently installed cadmium to my c330 and it works perfectly, could I get what you are doing?

**Reply by cdd100** [reply]
I build archlinuxARM for the acer r3 and it fails to boot. I will not install it to emmc.

**Reply by cdd100** [reply]
i might just use https://github.com/hexdump0815/imagebuilder

**Comment by chinh4** [comment]
Those images were built before support was added, you're gonna have to build it yourself.

**Reply by cdd100** [reply]
Are 0.4.0 pre2 images supported?

**Reply by cdd100** [reply]
I am a beginner and not sure how to build an image. I do have another laptop running Ubuntu 22.04 On a 4th gen core i5. (Acer travelmate b113). Are there any other arm distros I can try that could potentially work?

**Reply by chinh4** [reply]
> I am a beginner and not sure how to build an image. I do have another laptop running Ubuntu 22.04 On a 4th gen core i5. (Acer travelmate b113). Are there any other arm distros I can try that could potentially work?

There are none other arm based distros that install to the emmc. Arch Linux arm works with the Acer r13 version for the c330.

**Comment by cdd100** [comment]
I apologize for any inconvenience this may cause you, but how would I build my own debian image for the c330? Thanks in advance,
cdd100

**Reply by chinh4** [reply]
check the README
